### PR TITLE
Toolchain support for Android ARM64 targets

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@tvolkert @mravn-google

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -355,16 +358,23 @@ class AndroidDevice extends Device {
...
+    if (!(devicePlatform == TargetPlatform.android_arm ||
+          devicePlatform == TargetPlatform.android_arm64) &&
+        !debuggingOptions.buildInfo.isDebug) {
       printError('Profile and release builds are only supported on ARM targets.');
       return new LaunchResult.failed();
     }
 
+    BuildInfo buildInfo = debuggingOptions.buildInfo;
+    if (devicePlatform == TargetPlatform.android_arm64)
+      buildInfo = buildInfo.withTargetPlatform(TargetPlatform.android_arm64);
```
[the remark this answers is not in the document]
Comment (jason-simmons) [reply packages/flutter_tools/lib/src/android/android_device.dart]: This is for scenarios where we don't know the target platform until we discover what kind of device is attached

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 2]
```diff
@@ -355,16 +358,23 @@ class AndroidDevice extends Device {
     if (!await _checkForSupportedAdbVersion() || !await _checkForSupportedAndroidVersion())
       return new LaunchResult.failed();
 
-    if (await targetPlatform != TargetPlatform.android_arm && !debuggingOptions.buildInfo.isDebug) {
+    final TargetPlatform devicePlatform = await targetPlatform;
+    if (!(devicePlatform == TargetPlatform.android_arm ||
```
[the remark this answers is not in the document]
Comment (eseidelGoogle) [reply packages/flutter_tools/lib/src/android/android_device.dart]: Dart could support x64 if needed: https://github.com/flutter/flutter/issues/9253 however x86 AOT would be some months/weeks of work I'm told.

**Pull request merged**